makelib/golang.mk

```diff
@@ -132,15 +132,37 @@ endif
 GO_COMMON_FLAGS = $(GO_BUILDFLAGS) -tags '$(GO_TAGS)'
 GO_STATIC_FLAGS = $(GO_COMMON_FLAGS) $(GO_PKG_STATIC_FLAGS) -installsuffix static  -ldflags '$(GO_LDFLAGS)'
 
+GO_INIT_REQUIRED_TARGETS = go.vendor.lite
+GO_VENDOR_TARGET = go.vendor
+GO_VENDOR_CHECK_TARGET = go.vendor.check
+GO_VENDOR_UPDATE_TARGET = go.vendor.update
```

Comment:
these variables seem unnecessary now that you `go mod` and `dep` are mutually exclusive. I think it would be much simpler to just do:

```
go.vendor:
ifeq ($(GO111MODULE),on)
    <go mod stuff>
else
    <go mod stuff>
endif
```

Reply:
The main reason is the dependency that `go.vendor.lite`, `go.vendor.lite` and `go.vendor.lite` have on `$(DEP)`

If we use the format above, I'm not sure how easily we can associate the dependency of `$(DEP)` to the `<dep stuff>` inside the loop.

Reply:
good point. how about:

```
ifeq ($(GO111MODULE),on)
go.vendor:
    <go mod stuff>
else
go.vendor: $(DEP)
    <dep stuff>
endif
```

I find the variables with target names a bit harder to read.